### PR TITLE
Refactor Global.iPause -> Global.pause

### PR DESCRIPTION
**Pull request by carmel4a**
- rename
- pause is now private member, with public access by methods.
- one small scope fixup in DynObj.cpp

### CODE REVIEW

File: DynObj.cpp [hunk 1]
```diff
@@ -3446,7 +3446,6 @@ bool TDynamicObject::Update(double dt, double dt1)
                         MoverParameters->PantRearVolt = 0.0;
                 }
                 else {
-//                    Global.iPause ^= 2;
```
Comment (carmel4a) [inline DynObj.cpp]: Propably old unused comment code.

File: DynObj.cpp [hunk 2]
```diff
@@ -3840,8 +3839,6 @@ void TDynamicObject::TurnOff()
 // przeliczanie dźwięków, bo będzie słychać bez wyświetlania sektora z pojazdem
 void TDynamicObject::RenderSounds() {
 
-    if( Global.iPause != 0 ) { return; }
```
Comment (carmel4a) [inline DynObj.cpp]: Moved to `basic_section::update_sounds`

File: Globals.cpp [hunk 3]
```diff
@@ -566,7 +566,11 @@ global_settings::ConfigParse(cParser &Parser) {
             // czy po wczytaniu ma być pauza?
             Parser.getTokens();
             Parser >> token;
-            iPause |= (token == "yes" ? 1 : 0);
```
Comment (carmel4a) [inline Globals.cpp]: There is no longer a "paused on start" bitflag.

File: World.cpp [hunk 4]
```diff
@@ -555,18 +555,16 @@ void TWorld::OnKeyDown(int cKey) {
...
             }
-#endif
+            #endif
             // else if (cKey=='3') Global::iWriteLogEnabled^=4; //wypisywanie nazw torów
         }
-    }
+    } 
     else if( cKey == GLFW_KEY_ESCAPE ) {
-        // toggle pause
-        if( Global.iPause & 1 ) // jeśli pauza startowa
```
Comment (carmel4a) [inline World.cpp]: Delete "paused on start" check, and bitflag itself.

File: simulation.cpp [hunk 5]
```diff
@@ -58,8 +58,6 @@ state_manager::deserialize( std::string const &Scenariofile ) {
         // as long as the scenario file wasn't rainsted-created base file override
         Region->serialize( Scenariofile );
     }
-
```
Comment (carmel4a) [inline simulation.cpp]: That probably should be 0b1 not 0b10. Idk it is necessary here.

File: simulation.cpp [hunk 6]
```diff
@@ -58,8 +58,6 @@ state_manager::deserialize( std::string const &Scenariofile ) {
...
-    Global.iPause &= ~0x10; // koniec pauzy wczytywania
```
Comment (carmel4a) [inline simulation.cpp]: That probably should be 0b1 not 0b10. Idk it is necessary here.

File: Console.cpp [hunk 7]
```diff
@@ -284,19 +284,19 @@ void Console::ValueSet(int x, double y)
...
-                Global.iPause &= ~8;
-            }
-            else
-            { // błąd komunikacji - zapauzować symulację?
-                if (!(Global.iPause & 8)) // jeśli jeszcze nie oflagowana
-                    Global.iTextMode = GLFW_KEY_F1; // pokazanie czasu/pauzy
-                Global.iPause |= 8; // tak???
-                PoKeys55[0]->Connect(); // próba ponownego podłączenia
+    if((iMode == 4) && ( PoKeys55[0] )){
+        if( PoKeys55[0]->Update( Global.get_pause_flag( Global.PAUSES::LAGS ) ) ){
```
Comment (mrozigor) [inline Console.cpp]: Probably sticking to CamelCase would be better than mixing with underscores.

**Comment by carmel4a** [comment]
Closed PR due to in-project decision. Some features like deletion of first flag may be included later.

**Pull request closed without merging**